### ISSUE TITLE
ClassLoader leak in TomcatMetrics

### ISSUE THREAD

**Issue by rPraml** [micrometer-metrics/micrometer#1424]
When a web application is deployed multiple times, the old classloader will not be garbage collected. This is similar to #827 

I debugged with eclipse and see the following behaviour:
registerMetricsEventually will be called 2x (for "type", "StringCache" and "j2eeType", "Servlet") but on redeploy, the listener, which removes itself with `mBeanServer.removeNotificationListener` is only called for the "j2eeType", "Servlet" filter. 
You see also in the debug screenshot, that there are more than one "type", "StringCache" refenece in the listenerList.

Tomcat version is 9.0.20 (tested on windows)
 
Possible fix: Let TomcatMetrics implement DisposableBean and remove all listeners on destroy.

(Is there a simple workaround to disable the TomcatMetrics?)

jprofiler
![grafik](https://user-images.githubusercontent.com/5530228/57813993-7ab26700-7772-11e9-8b6d-770ba2cf2be3.png)

eclipse debug
![grafik](https://user-images.githubusercontent.com/5530228/57814015-8ef66400-7772-11e9-8ba5-b61c0cba905c.png)

**Comment by rPraml** [comment]
It seems, that micrometer adds a listener until the "StringCache" bean is registered.
The listener is removed on that event.
Unfortunately, the listener will not be removed, when the "StringCache"-register event does never occur.

I tried to create a patch https://github.com/FOCONIS/micrometer/commit/e785d71614064bf5da1312a7b5ebbd81a60c6ac0 but I noticed that the changed file (TomcatMetricsBinder) is not used by Spring boot 2 any more.

For now, I have disabled this metric with `@EnableAutoConfiguration(exclude = TomcatMetricsAutoConfiguration.class)`
